src/pmr2/wfctrl/cmd.py

```diff
@@ -183,34 +190,41 @@ def read_remote(self, workspace, target_remote=None, **kw):
     def write_remote(self, workspace, target_remote=None, **kw):
         target_remote = target_remote or self.default_remote
         stdout, err, return_code = self.execute(*self._args(workspace, 'remote',
-            'rm', target_remote))
+                                                            'rm', target_remote))
         stdout, err, return_code = self.execute(*self._args(workspace, 'remote',
-            'add', target_remote, self.remote))
+                                                            'add', target_remote, self.remote))
 
     def pull(self, workspace, username=None, password=None, **kw):
         # XXX origin may be undefined
         target = self.get_remote(workspace,
-            username=username, password=password)
+                                 username=username, password=password)
         # XXX assuming repo is clean
         args = self._args(workspace, 'pull', target)
-        return self.execute(*args)
+        output = self.execute(*args)
+        return_code = 0
+        if output[1]:
+            return_code = 1
+        return '\n'.join(output[0]).encode(), output[1], return_code
 
     def push(self, workspace, username=None, password=None, branches=None,
-            **kw):
+             **kw):
         """
         branches
             A list of branches to push.  Defaults to --all
         """
-
         push_target = self.get_remote(workspace,
-            username=username, password=password)
+                                      username=username, password=password)
         args = self._args(workspace, 'push', push_target)
         if not branches:
             args.append('--all')
         elif isinstance(branches, list):  # pragma: no cover
             args.extend(branches)
 
-        return self.execute(*args)
+        output = self.execute(*args)
+        return_code = 0
+        if output[1]:
+            return_code = 1
+        return '\n'.join(output[0]).encode(), output[1], return_code
```

Comment:
The `Popen` return code is captured in the third element (idx 2) of the tuple as per https://github.com/PMR2/pmr2.wfctrl/blob/master/src/pmr2/wfctrl/core.py#L304-L305  - just use that instead?  Avoids the condition check and thus the coverage issue.

Reply:
Done.